sphinx/cmd/make_mode.py

```diff
@@ -106,7 +106,33 @@ def build_latexpdf(self) -> int:
             raise RuntimeError('Invalid $MAKE command: %r' % makecmd)
         try:
             with chdir(self.builddir_join('latex')):
-                return subprocess.call([makecmd, 'all-pdf'])
+                if '-Q' in self.opts:
+                    with open('__LATEXSTDOUT__', 'w') as outfile:
+                        returncode = subprocess.call([makecmd,
+                                                      'all-pdf',
+                                                      'LATEXOPTS=-halt-on-error',
+                                                      ],
+                                                     stdout=outfile,
+                                                     stderr=subprocess.STDOUT,
+                                                     )
+                    if returncode:
+                        print('Latex error: check %s' %
+                              self.builddir_join('latex', '__LATEXSTDOUT__')
+                              )
+                elif '-q' in self.opts:
+                    returncode = subprocess.call(
+                        [makecmd,
+                         'all-pdf',
+                         'LATEXOPTS=-halt-on-error -interaction=batchmode',
+                         ],
+                    )
```

Comment:
This is about ok with my Latexmk from TeXLive 2024.  But with Latexmk from TeXLive 2019 (`This is Latexmk, John Collins, 26 Dec. 2019, version: 4.67`.), after the first pdflatex/xelatex/lualatex in batchmode which is completely silent, Latexmk extracts from the Latex log all lines saying some reference or citation is undefined.  And there are often lots of them on first run when auxiliary files do not exist.  For example, with the SymPy project this makes currently for about 8400 lines of console output! With `This is Latexmk, John Collins, 7 Apr. 2024. Version 4.85` there is no such output.

Thus the method of running the *latex in batchmode but Latexmk itself normally is not really "quiet" on TeXLive 2019 at least.  I will thus change the commit in favor of passing `-silent` (which is same as its `-quiet`) option to Latexmk, although I find it too silent for diagnostics.

```suggestion
                elif '-q' in self.opts:
                    returncode = subprocess.call(
                        [makecmd,
                         'all-pdf',
                         'LATEXOPTS=-halt-on-error',
                         'LATEXMKOPTS=-silent',
                         ],
                    )
```

Reply:
Currently it thus impossible to do things such as `make latexpdf O=-q LATEMKOPTS=-xelatex`  for a project using `xelatex`.  One will have to do `make latex O=-q` (or equivalent with sphinx-build) and in second step do `make -C <builddir> all-pdf LATEXMKOPTS=-xelatex -silent` to achieve similar effect.